src/adapt.ts

```diff
@@ -18,6 +21,77 @@ function toNumericString(value: { toString: () => string }) {
     return BigInt(value.toString()).toString();
 }
 
+function isNamedTuple(type: string) {
+    return type.includes(NAMED_TUPLE_DELIMITER);
+}
+
+function isTuple(type: string) {
+    return type[0] === "(" && type[type.length - 1] === ")";
+}
+
+// Can't use String.split since ':' also can be inside type
+// Ex: x : (y : felt, z: SomeStruct)
+function parseNamedTuple(namedTuple: string): starknet.Argument {
+    const index = namedTuple.indexOf(NAMED_TUPLE_DELIMITER);
+    const name = namedTuple.substring(0, index);
+    const type = namedTuple.substring(name.length + NAMED_TUPLE_DELIMITER.length);
+
+    return { name, type };
+}
+
+// Returns types of tuple
+function extractMemberTypes(s: string): string[] {
+    // Replace all nested tuples with '#'
```

Comment:
Can it say `Replace all top-level tuples with '#'`? I spent some time trying to figure out what it does with all the tuples of tuples of tuples, only to realize it doesn't do anything.

Reply:
My bad. Replaced with suggested string